### PR TITLE
Multisig improvements

### PR DESCRIPTION
**Pull request by dorin-iancu**
- group ID for actions, allows querying for all actions in a group
- allows specifying custom gas limit for action
- `signBatch` endpoint, which allows signing multiple actions at once
- `signAndPerform` endpoint, which allows signing and performing an action in the same call. Also added for `signBatchAndPerform` endpoint
- moved sign actions to a separate file
- added custom types for `ActionId`, `GroupId`, and `UserId` - Only for readability purposes

### CODE REVIEW

File: contracts/multisig/src/multisig_perform.rs [hunk 1]
```diff
@@ -177,15 +190,18 @@ pub trait MultisigPerformModule:
                 if let Result::Err(e) = result {
                     sc_panic!(e);
                 }
+
                 OptionalValue::None
-            },
+            }
             Action::SendAsyncCall(call_data) => {
-                let gas_left = self.blockchain().get_gas_left();
+                let gas = call_data
```
Comment (ccorcoveanu) [inline contracts/multisig/src/multisig_perform.rs:197]: If I think about it, I am a bit worried on prioritising the call_data gas limit. The thing is, that if for some reason (increased storage on the contract we are calling) the saved gas limit is no longer enough, this action cannot be executed anymore. 
The only solution I see is keeping the gas_limit only as an extra info. That way the front-end can simulate, or increase it if it fails the first time.

File: contracts/multisig/src/multisig_perform.rs [hunk 2]
```diff
@@ -118,12 +124,17 @@ pub trait MultisigPerformModule:
         // syntax aside, the async_call_raw kills contract execution so cleanup cannot happen afterwards
         self.clear_action(action_id);
 
+        let group_id = self.group_for_action(action_id).take();
```
[the remark this answers is not in the document]
Comment (dorin-iancu) [reply contracts/multisig/src/multisig_perform.rs]: Done.

File: contracts/multisig/src/multisig_perform.rs (hunk 1 above)
Comment (dorin-iancu) [reply contracts/multisig/src/multisig_perform.rs:197]: Well, if that happens, they can simply propose another time. It should be a pretty rare case, I wouldn't complicate the logic of the contract with additional arguments just for this edge case.

Comment (ccorcoveanu) [reply contracts/multisig/src/multisig_perform.rs:197]: ok - let's leave it like this and see if we hit this issue along the way

ps - I was not proposing to add another argument. I was proposing to remove this check and only use get_gas_left.
The frontend will use the saved gas_limit as a reference

**Review by ccorcoveanu — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Pull request merged**